### ISSUE TITLE
Argument parsing semantics changed, add an example to README

### ISSUE THREAD

**Issue by Fuuzetsu** [asayers/patiently#2]
This used to work

```
patiently -j 192 -- '/nix/store/ywm8z6pmvmh7vlq0a8ig6mjvrvkw31ry-rust_sim-0.1.0/bin/sim --metrics disable --no-snapshots --config <(echo "contrib: { loss_tolerance: -0.006 }") /tmp/20160912.ordercache'
```

But now it gives no such file or directory after updating to master. Presumably changing from clap to bpaf broke it.

It seems it now needs to be

```
patiently -j 192 -- /nix/store/ywm8z6pmvmh7vlq0a8ig6mjvrvkw31ry-rust_sim-0.1.0/bin/sim --metrics disable --no-snapshots --config <(echo "contrib: { loss_tolerance: -0.006 }") /tmp/20160912.ordercache
```

without the quotes: which is fine, but maybe README should have a better example

**Comment by asayers** [comment]
That's right, we used to (unconditionally) spawn a shell and pass the arguments to that for interpretation. Now the arguments are simply run. This means if you want to use pipes and redirections you need to explicitly include an `sh -C` yourself.

**Comment by asayers** [comment]
What kind of example did you have in mind? I did update the README when I made this change to get rid of the quotes in all the example invocations.

**Comment by Fuuzetsu** [comment]
>What kind of example did you have in mind? I did update the README when I made this change to get rid of the quotes in all the example invocations.

I guess some example where a `--some-flag` is involved so that `--` is required. `patiently -j 2 foo --flag-for-foo` doesn't work at the moment as `--flag-for-foo` ends up being considered a `patiently` flag. `patiently -j 2 -- foo --flag-for-foo` is fine of course.
